index.js

```diff
@@ -321,6 +323,39 @@ const captureWebsite = async (input, options) => {
 		await options.beforeScreenshot(page, browser);
 	}
 
+	if (screenshotOptions.fullPage) {
+		// Get the height of the rendered page
+		const bodyHandle = await page.$('body');
+		const bodyBoundingHeight = await bodyHandle.boundingBox();
+		await bodyHandle.dispose();
+
+		// Scroll one viewport at a time, pausing to let content load
+		const viewportHeight = viewportOptions.height;
+		let viewportIncrement = 0;
+		while (viewportIncrement + viewportHeight < bodyBoundingHeight) {
+			const navigationPromise = page.waitForNavigation();
```

Comment:
Shouldn't this line wait for `networkidle` instead of the default [`loaded`]?

[`loaded`]: https://github.com/puppeteer/puppeteer/blob/master/docs/api.md#pagewaitfornavigationoptions

Reply:
I probably agree with you, but care to clarify?

Reply:
> I probably agree with you, but care to clarify?

Waiting for network idle allows lazy elements to load (elements loaded after the "load" event of the document). I suggest doing as follows:

1. Wait "load" event of the document.
2. Scroll the page to its maximum Y (this will load lazy elements like lazy loaded images).
3. Wait for network idle.

Reply:
It's also worth to keep in mind that some (old) websites can actually never fulfill the requirements for the network idle navigation event - for example frequent chat requests.

In the end, the propsed flow above should work just fine.